compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -2607,7 +2607,7 @@ declare_lint! {
     /// [issue #73333]: https://github.com/rust-lang/rust/issues/73333
     /// [`Copy`]: https://doc.rust-lang.org/std/marker/trait.Copy.html
     pub CENUM_IMPL_DROP_CAST,
-    Warn,
+    Deny,
```

Comment:
This looks like it should be a separate PR?

Also, should this be shown in the future compat report as well so people notice when their dependencies still rely on this?

Reply:
This does not affect dependencies in a *new* way. We already did report this in future compat reports. All I did change the level in main crates.

Reply:
> This looks like it should be a separate PR?

Also this PR flips the logic *again* on what happens to casts from c-like enums with Drop impls. I found it prudent to bump the level :)

Reply:
> We already did report this in future compat reports.

I don't think so. This requires a special flag in the `FutureIncompatibleInfo`. See e.g. https://github.com/rust-lang/rust/pull/95372.

Reply:
> Also this PR flips the logic again on what happens to casts from c-like enums with Drop impls. I found it prudent to bump the level :)

Sure, I agree. I am just saying that's a decision process which should probably be separate from the rest of this PR.